pkg/publicapi/client/v2/options.go

```diff
@@ -37,11 +40,47 @@ type Options struct {
 
 	// Headers is a map of headers to add to all requests.
 	Headers http.Header
+
+	// TLSOptions provides info on how we want to use TLS
+	TLS TLSOptions
+}
+
+type TLSOptions struct {
+	// UseTLS denotes whether to use TLS or not
+	UseTLS bool
+	// Insecure activates TLS but does not verify any certificate
+	Insecure bool
+	// CACert specifies the location of a self-signed CA certificate
+	// file
+	CACert string
```

Comment:
Would it be possible to trim this down to two fields and remove the `UseTLS` field? I think `UseTLS` is used to determine the URI schema, right? Could the schema be inferred from the other two values?

Reply:
It can't be determined unfortunately because the Insecure bool being false doesn't infer any knowledge of whether to use TLS connections or not.  e.g. Insecure=True could be interpreted as use TLS but don't verify, but Insecure=False would mean use TLS and verify and that would be the default value.